### ISSUE TITLE
Running the code

### ISSUE THREAD

**Issue by stellaCL** [sfarrens/sf_deconvolve#1]
Hi!

Just a simple question here: how do you run this code?

I tried the example syntax shown in "Execution" inside IPython and python and it always prints me the error "SyntaxError: invalid syntax". I did not use the example files, but my own image.fits and psf.fits, which according to the documentation should work.

Perhaps there is something I missed from the documentation?

C.

**Comment by sfarrens** [comment]
Hi,

You may have an issue running the main script inside iPython as it is looking for command line arguments. If you run the same execution command in a terminal with your input files it should work. If you have any further issues let me know. :)

Cheers,
Sam

**Comment by stellaCL** [comment]
I can only run this on my Mac using
`> python sf_deconvolve.py -i input.fits -p psf.fits -o test`
which does execute the script, but then I get the following error message
`ERROR: global name 'data' is not defined`

**Comment by sfarrens** [comment]
Is the log file being generated? If so, can you paste the contents here so that I can spot where the error is occurring?

**Comment by stellaCL** [comment]
> 31/05/2017 11:58:27 The log file has been set-up.
> 31/05/2017 11:58:27  Running SF_DECONVOLVE
> 31/05/2017 11:58:27 ERROR: global name 'data' is not defined
> Traceback (most recent call last):
>   File "sf_deconvolve.py", line 221, in main
>     run_script(log)
>   File "sf_deconvolve.py", line 102, in run_script
>     opts.current_res)
>   File "/Users/myhome/WORK/VISIR/sf_deconvolve-master/lib/file_io.py", line 146, in read_input_files
>     input_data = data.reshape(1, *input_data.shape)
> NameError: global name 'data' is not defined

**Comment by sfarrens** [comment]
Sorry for the delay. I have committed a bug fix, please pull the update and try to run the code again.

**Comment by stellaCL** [comment]
Sorry for the delay. It seems to be working now. 

But now if the input FITS files have more than one extension, then I get the message
`ERROR: The number of input images must be greater than or or equal to the number of PSF images.`

I also cannot execute sf_deconvolve "inside" python, but only by using the syntax I mentioned two posts ago, which means that I would have to extract the appropriate FITS extension into a new image and then run sf_deconvolve.

**Comment by sfarrens** [comment]
Hi, 

I have updated the top-level readme with some slightly improved execution instructions. Here are the main points (your error is most likely related to the third point):

- The code is currently set up to read a 3D array of images where element [0] would be the first image and element [1] the second, etc. So a set of 10 images with 41x41 pixels should have shape [10, 41, 41]. I have added example fits files with this format to the example directory.

- If you have a fixed PSF (i.e. it is the same for every image) then the PSF array need only be 2D. e.g.  for a 13x13 pixel PSF the shape should be [13, 13].

- If you have a spatially varying PSF (i.e. it is different for every image) then the PSF array should have the same number of elements as the image array. e.g. 10 images should have 10 PSFs which could have shape [10, 13, 13]. You will get the error you reported if the first element of the shape of the PSFs is greater than that of the images.

- If you want to run the code in an active Python session you will need to manually define a dictionary with the default parameter values and create two Numpy arrays; one with your input images and one with your PSFs. See the updated readme.md. 

Let me know if this helps! :)
